framework/SupervisedLearning/SyntheticHistory.py

```diff
@@ -129,7 +129,10 @@ def __evaluateLocal__(self, featureVals):
       targets = settings['target']
       indices = tuple(self.target.index(t) for t in targets)
       params = self.trainedParams[algo]
-      signal = algo.generate(params, pivots, settings)
+      if algo.canGenerate():
+        signal = algo.generate(params, pivots, settings)
+      else:
+        signal = algo.signal; 
```

Comment:
This might need thinking about. I'm not sure how much we want to just return the original signal if it can't generate. Technically if a TSA can't generate, it shouldn't be usable by SyntheticHistory.

I wonder if we need a Postprocessor that acts like the SyntheticHistory but only extracts characteristics and puts them into a DataObject for each realization. This would allow testing characterizers rather than generators.